packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -142,109 +94,493 @@ describe("Decentraland", () => {
+
+          await executeOrderAction.waitForNonceToUpdate();
+
+          await confirmBalances(land, [GNOSIS_SAFE_ADDRESS], [1]);
+        });
+      });
 
-        await confirmBalances(landContract, [ephemeralAddress], [1]);
+      // Allowence-of-allowence doesn't work
+      // See more: https://github.com/tasitlabs/TasitSDK/issues/273
+      describe.skip("Using an ephemeral wallet allowed to spend Gnosis Safe wallets' funds", () => {
+        beforeEach("onboarding", async () => {
+          // Funding ephemeral account with some ethers to pay for gas
+          // TODO: ephemeralWallet should broadcast this action
+          const toAddress = ephemeralAddress;
+          gnosisSafe.setSigners([gnosisSafeOwner]);
+          gnosisSafe.setWallet(gnosisSafeOwner);
+          const transferEthersAction = gnosisSafe.transferEther(
+            toAddress,
+            SMALL_AMOUNT
+          );
+          await transferEthersAction.waitForNonceToUpdate();
+          await confirmEtherBalances(provider, [toAddress], [SMALL_AMOUNT]);
+
+          // Gnosis Safe should approve ephemeral account to spend its MANA Tokens
+          // TODO: ephemeralWallet should broadcast this action
+          const contractAddress = mana.getAddress();
+          const contractABI = mana.getABI();
+          const functionName = "approve";
+          const spender = ephemeralAddress;
+          const argsArray = [spender, manaAmountForShopping];
+          gnosisSafe.setSigners([gnosisSafeOwner]);
+          gnosisSafe.setWallet(gnosisSafeOwner);
+          const ephemeralApprovalAction = gnosisSafe.customContractAction(
+            contractAddress,
+            contractABI,
+            functionName,
+            argsArray
+          );
+          await ephemeralApprovalAction.waitForNonceToUpdate();
+          const owner = GNOSIS_SAFE_ADDRESS;
+          const ephemeralAllowance = await mana.allowance(owner, spender);
+          expect(`${ephemeralAllowance}`).to.equal(`${manaAmountForShopping}`);
+
+          // Ephemeral account approves Marketplace to spend their mana tokens (actually tokens from Gnosis Safe).
+          mana.setWallet(ephemeralWallet);
+          const marketplaceApprovalAction = mana.approve(
+            MARKETPLACE_ADDRESS,
+            manaAmountForShopping,
+            gasParams
+          );
+          await marketplaceApprovalAction.waitForNonceToUpdate();
+          const marketplaceAllowance = await mana.allowance(
+            ephemeralAddress,
+            MARKETPLACE_ADDRESS
+          );
+          expect(`${marketplaceAllowance}`).to.equal(
+            `${manaAmountForShopping}`
+          );
+        });
+
+        it("should buy an estate", async () => {
+          const {
+            assetId,
+            nftAddress,
+            seller,
+            priceInWei,
+            expiresAt,
+          } = estateForSale;
+
+          checkAsset(estate, mana, estateForSale, GNOSIS_SAFE_ADDRESS);
```

Comment:
Wouldn't the buyer - the 4th argument - be the ephemeral account here?

Reply:
Yes, the correct name for the 4th argument maybe is `ownerOfFunds` because `checkAsset` will check if that account has enough funds, and `ephemeralAccount` will not pass thru this test.
Change `checkAsset` to verify the allowed balance might be error prone because passing thru that expectation not guarantee that the buyer will be successful on the purchase.
Should I rename the 4th argument or add a comment? What do you think?

Reply:
Ah yes, that's an interesting point. I'm leaning towards renaming the 4th parameter, and also adding a comment in the code that talks about this briefly